galaxy wrapper docs with planemo

suhaibMo opened the pull request:
In this PR,
- Galaxy wrapper generation with planemo so that we and the wider community can participate in a consistent manner.

suhaibMo committed:
adding doc for galaxy wrapper with planemo

suhaibMo committed:
more usage of help section and versions added

pcm32 commented on docs/writing_planemo_galaxy_wrapper.rst:
```diff
@@ -0,0 +1,222 @@
+
+Installing bioconda packages
+
+Developing galaxy tools wrappers for a particular bioconda recipe requires installation of the package in the same environment.  
```
I thought that planemo/galaxy did this for you directly (and that you don't need to install the conda package for the tool manually). Any comments @nh3 @cartal ? Thanks

nh3 replied:
I think planemo/galaxy handles the installation of packages required to run the tools. However, to generate test-data for planemo to test against (and for general development convenience), one needs to be able to run the tools outside planemo/galaxy. For that, the required packages need to be installed, preferably using conda.

nh3 commented on docs/writing_planemo_galaxy_wrapper.rst:
```diff
@@ -0,0 +1,222 @@
+    </macros>
+    <expand macro="requirements" />
+    <command detect_errors="exit_code"><![CDATA[
+        ln -s '$matrix' matrix.mtx;
```
IUC suggests use `&&` instead of `;` at the end of each command to ensure errors are properly caught.

nh3 commented on docs/writing_planemo_galaxy_wrapper.rst:
```diff
@@ -0,0 +1,222 @@
+        scater-read-10x-results.R -d ./ -o '$R_scater_serialized'
+    ]]></command>
+    <inputs>
+        <param type="data" name="matrix" format="mtx" />
```
`"mtx"` seems not a galaxy supported format. Perhaps use `"txt"` for the moment?

nh3 commented on docs/writing_planemo_galaxy_wrapper.rst:
```diff
@@ -0,0 +1,222 @@
+
+Scater-read-10x-results.xml will have have generic input and input variable names. I’ve renamed for clarity. In addition, format of rds was renamed to `rdata` as it widely accepted within galaxy community. Symlinks were created to point input variable names. It is also recommended to use the version of bioconductor package used in bioconda to be in `tool id` version section for consistancy. For e ...
+
+<tool id="scater-read-10x-results" name="Scater read 10x data" version="0.0.3">
```
IUC suggests add a token in `macros.xml`:

```
<token name="@TOOL_VERSION@">0.0.3</token>
```

 and use it across wrappers:

```
version="@TOOL_VERSION@+galaxy0"
```

where `@TOOL_VERSION` is the version of the wrapped tool and `galaxy0` gives the version of the galaxy wrapper.

nh3 commented on docs/writing_planemo_galaxy_wrapper.rst:
```diff
@@ -0,0 +1,222 @@
+        <param type="data" name="barcodes" format="tabular" />
+    </inputs>
+    <outputs>
+        <data name="R_scater_serialized" format="rdata" label="" />
```
A useful label here might be `label="${tool.name} on ${on_string}: Rdata"`

nh3 commented on docs/writing_planemo_galaxy_wrapper.rst:
```diff
@@ -0,0 +1,222 @@
+            <param name="matrix" value="matrix.mtx"/>
+            <param name="genes" value="genes.tsv"/>
+            <param name="barcodes" value="barcodes.tsv"/>
+            <output name="R_scater_serialized" file="R_scater_serialized.rds"/>
```
Sometimes the binary output can be different for each run, e.g. scanpy AnnData objects seem to contain timestamp information. For those instances, one can use `compare="sim_size"` to check whether the size of the output is similar to expectation or not.

One can also add `ftype="rdata"` here to check the output format.

nh3 commented on docs/writing_planemo_galaxy_wrapper.rst:
```diff
@@ -0,0 +1,222 @@
+	Planemo test 
+All 1 test(s) successfully executed.
+This will create symlinks and use the input data provided within test environment and execute in galaxy server.
+
```
It might be useful to also include [Best Practices for Creating Galaxy Tools](https://galaxy-iuc-standards.readthedocs.io/en/latest/best_practices.html) and [Galaxy Tool XML Schema](https://docs.galaxyproject.org/en/latest/dev/schema.html) for reference.

pinin4fjords commented:
@nh3 - would you have time to resolve the issues you mention and then merge this PR? With @suhaibMo being away this probably won't get done in December otherwise and I'd like to add other docs :-)

pinin4fjords commented on docs/writing_planemo_galaxy_wrapper.rst:
```diff
@@ -0,0 +1,222 @@
+**************************
+Prerequisites
+**************************
+Installation of 
```
Unless this stuff is different from Planemo's documentation, we probably shouldn't repeat it here in case we go out of synch with them.

nh3 committed:
Update writing_planemo_galaxy_wrapper.rst

I've taken the liberty to directly editing this file. The edits are largely related to keeping in compliance of coding style recommended by IUC. Hopefully this is fine.

nh3 commented:
I've taken the liberty to directly edit this file. The edits are largely related to keeping in compliance of coding style recommended by IUC. Hopefully this is fine.

pinin4fjords commented:
Take all the liberties you want @nh3, you have a bit of experience in this now. Squash and merge when you're happy :-).
